File: Bislerium-Blogs.Server/Interfaces/IPagination.cs
using Bislerium_Blogs.Server.Enums;

namespace Bislerium_Blogs.Server.Interfaces
{
    public interface IPagination
    {
        public int? PageNumber { get; set; }
        public int? PageSize { get; set; }

        public string? Search { get; set; }

    }

}

File: bislerium-blogs.client/src/Interfaces/IPagination.ts
import { IBlog } from './Models/IBlog';

export enum SortBy {
  NEWEST,
  OLDEST,
  POPULARARITY,
  RANDOM,
}

export interface IPagination {
  pageNumber: number;
  pageSize: number;
  search?: string;
}

export interface IBlogPaginationDto extends IPagination {
  sortBy?: SortBy;
  ofThisSpecificMonth?: Date;
  tag?: string;
}

export interface IBlogPaginatedResponse {
  blogs: IBlog[];
  totalBlogs: number;
  currentPage: number;
}
